dataservice/api/study_file/resources.py

```diff
@@ -33,10 +33,34 @@ def get(self, after, limit):
         # Filter by study
         study_id = request.args.get('study_id')
         if study_id:
-            q = (q.filter_by(study_id=study_id))
+            q = q.filter(StudyFile.study_id == study_id)
+
+        pager = Pagination(q, after, limit)
+        keep = []
+        refresh = True
+        next_after = None
+        # Continue updating the page until we get a page with no deleted files
+        while (pager.total > 0 and refresh):
+            refresh = False
+            # Move the cursor ahead to the last valid file
+            next_after = keep[-1].created_at if len(keep) > 0 else after
+            # Number of results needed to fulfill the original limit
+            remain = limit - len(keep)
+            pager = Pagination(q, next_after, remain)
+
+            for st in pager.items:
+                merged = st.merge_indexd()
+                if merged is not None:
+                    keep.append(st)
+                else:
+                    refresh = True
+
+        # Replace original page's items with new list of valid files
+        pager.items = keep
+        pager.after = next_after if next_after else after
```

Comment:
Need a better way of abstracting this from the resource

Reply:
Could you make this a method in flask_indexd and then do: 

```
from dataservice.extensions.flask_indexd import paginate as paginate_indexd
```

```
# Indexd Pagination
pager = paginate_indexd(q, after, limit)

return (StudyFileSchema(many=True)
                .jsonify(pager))
```

Reply:
Could do that, but maybe it's better lived in `common.pagination` or some sort of `IndexdResource`?